crates/cxx-qt-lib/src/types/qcolor.rs

```diff
@@ -75,6 +76,25 @@ impl Default for QColor {
     }
 }
 
+impl fmt::Display for QColor {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let r = self.red();
+        let g = self.green();
+        let b = self.blue();
+        let a = self.alpha();
+        // very simple heuristic to use a light foreground if background is dark and vice versa
+        let fg = if (r + b + g) < 384 { 255 } else { 0 };
+        // Use terminal escape codes to **actually** print the color
```

Comment:
Wonder what happens on other terminals, like does this work only on a POSIX terminal or in Windows as well? Or just sh/bash/zsh? 🤔

Reply:
@ahayzen Nowadays true color support is available in pretty much every terminal emulator, see [here for a full list](https://github.com/termstandard/colors#fully-supporting).

Even the terminal application arch enemy (also known as **cmd.exe**) supports it.
The only notable terminal emulator without support is Terminal.app, but I am pretty sure that it fails silently there (i.e. the escape sequences are not shown but have no effect), so I don't find it necessar to implement true color support detection even though that would be very easy to do (see link above).

Reply:
There is Debug Vs Display ... "Debug should format the output in a programmer-facing, debugging context" sounds like it could be with colour in a terminal 🤔 Maybe we should see what other crates do  ...

Reply:
I think that sounds like a good compromise.
I have updated it to pretty-print with colors only in the Debug formatter and the Display formatter prints it without colors now.

Theoretically it would be also possible to detect if the output is a terminal (see `isatty(3)`), but let's not overengineer this and it would work only on Unix anyway.